[Samuel Tan] iP

tsammeow opened the pull request:
## Duke upgrades

- Add tasks
- List tasks
- Remove tasks
- *Look for tasks* 🔍

To run:
1. Clone this repo
2. Run `build` and `run` [Gradle](https://gradle.org/) scripts

Add more commands to the switch case:
```java
case "command-name":
    // your code here
    return "it is done";
```

To do:
- [x] Follow code guide
- [ ] Implement GUI

> Some quote

tsammeow committed:
duke level1

tsammeow committed:
duke level2

tsammeow committed:
level3

tsammeow committed:
level4

tsammeow committed:
textuitesting

tsammeow committed:
level5

tsammeow committed:
level6

tsammeow committed:
Fix string comparisons

tsammeow committed:
Follow style guide

tsammeow committed:
Prepare for saving

tsammeow committed:
Enable saving

tsammeow committed:
Merge branch 'branch-Level-7'

tsammeow committed:
Enable date support for Deadline and Event

tsammeow committed:
Merge branch 'branch-Level-8'

tsammeow committed:
Add Gradle to project

tsammeow committed:
Move files to Gradle

tsammeow committed:
Add Duke as Gradle main class

tsammeow committed:
Change list to use StringBuilder

tsammeow committed:
Organise into package

tsammeow committed:
Refactor code to use more OOP

tsammeow committed:
Update Task to follow OOP principles

tsammeow committed:
Add tests

tsammeow committed:
Add error handling for no input

tsammeow committed:
Make build script output jar file

tsammeow committed:
Make Gradle run script read from standard input

tsammeow committed:
Add find operation

tsammeow committed:
Make Storage actually use the provided file name

tsammeow committed:
Update code to follow coding standard

tsammeow committed:
Add javadoc to files

tsammeow committed:
Update Duke.java to follow coding standard

tsammeow committed:
Merge branch 'branch-Level-9'

tsammeow committed:
Merge branch 'branch-A-JavaDoc'

tsammeow committed:
Merge branch 'branch-A-CodingStandard'

zicotjia commented on app/src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+        this.deadline = LocalDate.parse(deadline);
+    }
+
+    @Override
+    public String getTaskIcon() {
+        return "D";
+    }
+
+    @Override
+    public String getDetails() {
+        return details + " (by: " + deadline.format(DateTimeFormatter.ofPattern("dd MMM yyyy")) + ")";
+    }
+
+    @Override
+    public String getEncodedDetails() {
+        return details + "|" + deadline;
+    }
+}
```
I like that you have this method to abstract out getting such details.

zicotjia commented on app/src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,138 @@
+    private static String handle(String command, String params) {
+        switch (command) {
+        case "list":
+            StringBuilder out = new StringBuilder();
+            for (int i = 0; i < tasks.size(); ++i) {
+                if (i != 0) {
+                    out.append("\n");
+                }
+                out.append(i + 1).append(". ").append(tasks.getTask(i));
+            }
+            if (out.toString().equals("")) {
+                return "No tasks left!";
+            }
+            return out.toString();
```
Perhaps a Command Class can be made to handle the cases?

zicotjia commented on app/src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,138 @@
+        parser = new Parser();
+        tasks = storage.load();
+
+        ui.respond("Hello! You have " + tasks.size() + " tasks. What can I do for you today?");
```
Having Duke tells the user about the number of tasks is a good QOL feature

zicotjia commented on app/src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,138 @@
+        case "mark":
+            int markedTask = checkTask(params);
+            if (markedTask < 0) {
+                return "Invalid task number!";
```
Another alternative is to create Exception subclass to handle invalid input and also store the error message. You could also make different subclass for different cases such as invalid date input, unknown command.

tsammeow committed:
Set up JavaFX

tsammeow committed:
Update Duke to allow different types of UI

tsammeow committed:
Change the way UI input works

tsammeow committed:
Implement Graphical UI

tsammeow committed:
Switch default UI to Graphical UI

tsammeow committed:
Merge branch 'branch-Level-10'

sunyuting2000 commented on app/src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,148 @@
+        }
+    }
+
+    private String handle(String command, String params) {
```
Good application of making the happy path prominent. Makes it easy to see what should happen should nothing go wrong. Nice!

sunyuting2000 commented on app/src/main/java/duke/DukeGui.java:
```diff
@@ -0,0 +1,21 @@
+    public void start(Stage stage) {
+        duke = new Duke("tasks.txt");
+        duke.setUi(new GraphicalUi(duke, stage));
+
```
Is there a reason for the empty line here?
Other than that good job!

sunyuting2000 commented on app/src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,57 @@
+            while (saved.hasNextLine()) {
+                try {
+                    tasks.add(Task.fromEncoded(saved.nextLine()));
+                } catch (DateTimeParseException ignored) {
```
Why are the exceptions ignored here? Perhaps try printing some error message?

sunyuting2000 commented on app/src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,114 @@
+     * @param s The encoded string.
+     * @return The created task.
+     */
+    public static Task fromEncoded(String s) {
```
Interesting way of decoding the input message to decide which task to add. Nice!

tsammeow committed:
Add assert statements for asumption checking

The code currently has no assertions.

Assertions help to document and catch important assumptions.

Let's add assertions to document our assumptions at various points in the code.

tsammeow committed:
Merge pull request #1 from tsammeow/branch-A-Assertions

Add assert statements for asumption checking

tsammeow committed:
Add support for shorter aliases for some commands

Commands can be long and annoying to type.

Let's add shorter aliases for some commonly used commands.

tsammeow committed:
Merge branch 'branch-C-FriendlierSyntax'

tsammeow committed:
Add plugin Shadow

tsammeow committed:
Use CheckStyle in Gradle

tsammeow committed:
Add Javadoc and reorder imports to conform to CheckStyle

tsammeow committed:
Merge branch 'branch-A-CheckStyle'

tsammeow committed:
Refactor MessageBox to only take in a string as input

tsammeow committed:
Merge branch 'master' into branch-A-CodeQuality

tsammeow committed:
Refactor GraphicalUi to follow SLAP

tsammeow committed:
Refactor Storage to reduce indentation level

tsammeow committed:
Refactor Duke to reduce indentation level

tsammeow committed:
Merge branch 'branch-A-CodeQuality'

tsammeow committed:
Add product screenshot

tsammeow committed:
Add user guide

tsammeow committed:
Bump javafx version

Fixes issue on some Linux environments, as per nus-cs2103-AY2223S1/forum#246

tsammeow committed:
Update README
